verification/verification_test.go

```diff
@@ -4,9 +4,26 @@ package verification
 
 import (
 	"flag"
+	"os"
+	"testing"
 )
 
-var sim_exe = flag.String("sim", "../simulator/target/debug/simulator", "path to simulator executable")
+var socket_exe *string
+var isEmulator *bool
+
+// This will be called before running tests, and it assigns the socket path based on command line flag.
+func TestMain(m *testing.M) {
+	isEmulator = flag.Bool("emulator", false, "socket type - emulator")
```

Comment:
Instead of a boolean flag, it would be nice to have a test target enum, where `emulator` and `simulator` are the current two options (and the default is simulator)